### ISSUE TITLE
GURA08a should not warn here.

### ISSUE THREAD

**Issue by JohanLarsson** [GuOrg/Gu.Roslyn.Asserts#146]
```cs
    [TestFixture(typeof(CallbackAnalyzer))]
    [TestFixture(typeof(PropertyMetadataAnalyzer))]
    public static class Valid<T>
        where T : DiagnosticAnalyzer, new()
    {
        private static readonly DiagnosticAnalyzer Analyzer = new T();
```
